kml/edit-kml-ground-overlay/src/main/java/com/esri/samples/edit_kml_ground_overlay/EditKMLGroundOverlaySample.java

```diff
@@ -0,0 +1,162 @@
+
+      // create a scene and add it to the scene view
+      ArcGISScene scene = new ArcGISScene(Basemap.createImagery());
+      sceneView = new SceneView();
+      sceneView.setArcGISScene(scene);
+
+      // create a geometry for the ground overlay
+      Envelope overlayGeometry = new Envelope(-123.066227926904, 44.04736963555683, -123.0796942287304, 44.03878298600624, SpatialReferences.getWgs84());
+
+      // create a KML icon for the overlay image
+      String overlayImageURI = new File("./samples-data/images/1944.jpg").getAbsolutePath();
+      KmlIcon overlayImage = new KmlIcon(overlayImageURI);
+
+      // create the KML ground overlay
+      KmlGroundOverlay kmlGroundOverlay = new KmlGroundOverlay(overlayGeometry, overlayImage);
+
+      // set the rotation of the ground overlay
+      kmlGroundOverlay.setRotation(-3.046024799346924);
+
+      // create a KML dataset with the ground overlay as the root node
+      KmlDataset kmlDataset = new KmlDataset(kmlGroundOverlay);
+
+      // create a KML layer using the KML dataset
+      KmlLayer kmlLayer = new KmlLayer(kmlDataset);
+
+      // add the KML layer to the scene view
+      sceneView.getArcGISScene().getOperationalLayers().add(kmlLayer);
+
+      // set the viewpoint to the ground overlay
+      sceneView.setViewpoint(new Viewpoint(kmlGroundOverlay.getExtent(), new Camera(kmlGroundOverlay.getGeometry().getExtent().getCenter(), 1250, 45, 60, 0)));
+
+      // create a slider for adjusting the overlay opacity
+      Slider slider = new Slider(0, 1, 1);
+      slider.setMaxWidth(300);
+
+      // add an event handler to the slider to apply opacity to the KML ground overlay based on the slider value
+      slider.setOnMouseReleased(event -> updateGroundOverlayOpacity(kmlGroundOverlay, slider.getValue()));
+      slider.setOnMouseDragged(event -> updateGroundOverlayOpacity(kmlGroundOverlay, slider.getValue()));
+
+      // create a controls box
+      VBox controlsVBox = new VBox();
+      controlsVBox.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(0,0,0,0.5)"), CornerRadii.EMPTY,
+              Insets.EMPTY)));
+      controlsVBox.setPadding(new Insets(10.0));
+      controlsVBox.setSpacing(10);
+      controlsVBox.setMaxSize(250, 20);
+      controlsVBox.getStyleClass().add("panel-region");
+
+      // create a label for the slider
+      Label label = new Label("Overlay Opacity: ");
+      // add the slider and label to the controls box
+      controlsVBox.getChildren().addAll(label, slider);
+
+      // add the scene view and controls to the stack pane
+      stackPane.getChildren().addAll(sceneView, controlsVBox);
+      StackPane.setAlignment(controlsVBox, Pos.TOP_LEFT);
+      StackPane.setMargin(controlsVBox, new Insets(10, 0, 0, 10));
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Updates the color of a KmlGroundOverlay with a provided alpha value.
```

Comment:
to be more accurate to what this method is doing, we should change this to:

"Changes the opacity of a KmlGroundOverlay using a provided alpha value."